builder/builder-next/builder.go

```diff
@@ -328,27 +335,44 @@ func (b *Builder) Build(ctx context.Context, opt backend.BuildConfig) (*builder.
 		frontendAttrs["ulimit"] = ulimits
 	}
 
+	reposAndTags, err := sanitizeRepoAndTags(opt.Options.Tags)
```

Comment:
Could be moved to `if b.useSnapshotter` condition since it's only used there. Obviously it's been awhile since I delved into this code but I must say I don't quite understand what's different with containerd snapshotter that requires this sanitizing that's not needed without containerd snapshotter.

Reply:
Yes, looks like this can be moved.

I see some issues with the (use of) `sanitizeRepoAndTags` though;


- We call this function already in `Backend.Build()` before we even hit `builder.Build`; that one _validates_ the values (but doesn't update the original value), so perhaps we should change that; https://github.com/moby/moby/blob/6eab4f55faf77bb0f253553d7d6fadb2ef6f4669/api/server/backend/build/backend.go#L57-L73
- It looks like we now have _three_ implementations of this same function;
    ```bash
    git grep 'func sanitizeRepoAndTags'
    api/server/backend/build/tag.go:func sanitizeRepoAndTags(names []string) ([]reference.Named, error) {
    builder/builder-next/builder.go:func sanitizeRepoAndTags(names []string) ([]reference.Named, error) {
    builder/builder-next/control/control.go:func sanitizeRepoAndTags(names []string) ([]reference.Named, error) {
    ```
- The one in `builder/builder-next/control/control.go` (`Controller.Solve()`) splitting this string again, only to validate (again) what we generated here
- `exporterAttrs` is overwritten further down (in case `len(opt.Options.Outputs) == 1`). Not sure immediately if it should _overwrite_ or _append_ options (need to check).